src/main/java/it/gov/pagopa/timeline/service/TimelineServiceImpl.java

```diff
@@ -86,6 +92,26 @@ public void sendToQueue(QueueOperationDTO queueOperationDTO) {
   public void saveOperation(QueueOperationDTO queueOperationDTO) {
     long startTime = System.currentTimeMillis();
 
+    if (TimelineConstants.CHANNEL_QRCODE.equals(queueOperationDTO.getChannel())
+    && TimelineConstants.OPERATION_TYPE_TRX.equals(queueOperationDTO.getOperationType())) {
+      Optional<Operation> existingOperation = timelineRepository.findByTransactionId(
+          queueOperationDTO.getTransactionId());
+
+      if (existingOperation.isPresent()) {
+        Operation operation = existingOperation.get();
+        if (ignoreCombinations.contains(queueOperationDTO.getStatus() + operation.getStatus())) {
+          return;
+        }
+        else if (queueOperationDTO.getStatus().equals(TimelineConstants.TRX_STATUS_REWARDED) && operation.getStatus().equals(TimelineConstants.TRX_STATUS_AUTHORIZED)) {
```

Comment:
in order to avoid to handle null, always write the constant as first member of a equals
```suggestion
        else if (TimelineConstants.TRX_STATUS_REWARDED.equals(queueOperationDTO.getStatus().) && TimelineConstants.TRX_STATUS_AUTHORIZED.equals(operation.getStatus())) {
```